mkosi/distributions/arch.py

```diff
@@ -0,0 +1,190 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+import os
+import platform
+from pathlib import Path
+from textwrap import dedent
+from typing import List, Set
+
+from ..backend import (
+    MkosiPrinter,
+    MkosiState,
+    OutputFormat,
+    add_packages,
+    complete_step,
+    disable_pam_securetty,
+    run,
+    sort_packages,
+)
+from ..mounts import mount_api_vfs
+from . import DistributionInstaller
+
+
+class ArchInstaller(DistributionInstaller):
+    @classmethod
+    def cache_path(cls) -> List[str]:
+        return ["var/cache/pacman/pkg"]
+
+    @classmethod
+    def install(cls, state: "MkosiState") -> None:
+        return install_arch(state)
+
+
+def invoke_pacman(root: Path, pacman_conf: Path, packages: Set[str]) -> None:
+    run(["pacman", "--config", pacman_conf, "--noconfirm", "-Sy", *sort_packages(packages)], env=dict(KERNEL_INSTALL_BYPASS="1"))
+
+
+@complete_step("Installing Arch Linux…")
+def install_arch(state: MkosiState) -> None:
```

Comment:
Will do. I mostly did it in two different ways for Arch and Debian to showcase how it can work.

Reply:
I decided against this for now, since this PR is already unwieldy enough and I think it makes it easier to compare without the additional change in indentation.